api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1758,15 +1768,14 @@ type AzurePlatformSpec struct {
 	// +immutable
 	VnetID string `json:"vnetID,omitempty"`
 
-	// SubnetName is the name of an existing subnet in VnetName / VnetID where Azure CCM will find an existing load
-	// balancer to be used for node egress.
-	//
-	// Example: if your SubnetID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>/subnets/<subnetName>,
-	//          your SubnetName is <subnetName>
+	// SubnetID is the subnet ID of an existing subnet where the load balancer for node egress will be created. This
+	// subnet ID is expected to be a subnet within the VnetID / VnetName.
+	// In ARO HCP, managed services will create the aforementioned load balancer.
 	//
-	// +kubebuilder:default:=default
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="SubnetID is immutable"
 	// +kubebuilder:validation:Required
-	SubnetName string `json:"subnetName"`
+	// +required
+	SubnetID string `json:"subnetID"`
```

Comment:
Here I would document the assumptions / requirements that are being made.

* The Subscription part specified in the VnetID has to match the value specified in the SubscriptionID attribute. If that's not the case let me know
* Does the resource group name part of the VnetID needs to match the ResourceGroupName value or the NetworkResourceGroupName value?
* Does the VNet name part of the SubnetID needs to match the value specified in `VnetName` or the resource name part in the `VnetID` attribute?

Reply:
> The Subscription part specified in the VnetID has to match the value specified in the SubscriptionID attribute. If that's not the case let me know

I added documentation for this.

Reply:
> Does the resource group name part of the VnetID needs to match the ResourceGroupName value or the NetworkResourceGroupName value?

Added documentation addressing this.